src/entities/createPool/types.ts

```diff
@@ -0,0 +1,33 @@
+import { Address, Hex } from 'viem';
+
+export interface CreatePoolBase {
+    buildCall(input: CreatePoolInput): CreatePoolBuildCallOutput;
+}
+
+export type CreatePoolInput = WeightedCreatedPoolInput;
+
+export type WeightedCreatedPoolInput = {
+    name?: string;
+    symbol: string;
+    tokens: Address[];
+    weights?: { tokenAddress: Address; weight: number }[];
```

Comment:
Out of curiosity, don't you loose some precision if you set `weight` as `number` since it needs to have 18 decimals?

Reply:
JS float number accepts up to 16 decimals, so users would lose the 17th and 18th, but I don't think this can be a problem, it's a very edge case, but a solution that could solve this precision problem and still deliver the benefits of the natural numbers is to use weight values like w1=80, w2=20, instead of w1=0.8, w2=0.2, for a 80/20 pool, for example

Reply:
Another question - is it ok to make `weights` optional when creating a Weighted pool? I think this should be required. How was that being handled in the previous SDK?

Reply:
1 - In the previous SDK was mandatory, I just didn't make mandatory to reduce the complexity of the pool creation in the equally weights cases, you found confusing for the user to create a pool without passing weights? I made this way, with equally distribution of weights as default, because we have a lot of 50/50 pools.

About the weight being a string, you're talking about a string with the solidity format, like "500....00" or just 0.5?

Obs.: Now that you pointed it out, I think that the rateProvider being optional (zeroAddress by default) may not be a good idea as well, I don't know if it's common to use zeroAddress as the rateProvider, or if this is just for tests, what do you think?